bleak/backends/bluezdbus/manager.py

```diff
@@ -946,36 +971,40 @@ def _parse_msg(self, message: Message):
 
                 if interface == defs.DEVICE_INTERFACE:
                     # handle advertisement watchers
+                    device_path = message_path
 
                     self._run_advertisement_callbacks(
-                        message.path, cast(Device1, self_interface), changed.keys()
+                        device_path, cast(Device1, self_interface), changed.keys()
                     )
```

Comment:
This could be indexed by device when the user has multiple bluetooth adapters it has to reject callbacks for the wrong device. But that should be another PR